suppression-backup-service/api/api_test.go

```diff
@@ -0,0 +1,120 @@
+package api_test
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path"
+	"sync"
+	"testing"
+
+	"github.com/phayes/freeport"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/suppression-backup-service/api"
+	"github.com/rudderlabs/rudder-server/suppression-backup-service/model"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	"github.com/stretchr/testify/require"
+)
+
+// exportPath creates a tmp dir and returns the path to it
+func exportPath() (baseDir string, err error) {
+	tmpDir, err := misc.CreateTMPDIR()
+	if err != nil {
+		return "", fmt.Errorf("could not create tmp dir: %w", err)
+	}
+	baseDir, err = os.MkdirTemp(tmpDir, "export")
+	if err != nil {
+		return "", fmt.Errorf("could not create tmp dir: %w", err)
+	}
+	return
+}
+
+func TestDataplaneAPIHandler(t *testing.T) {
+	config.Reset()
+	logger.Reset()
+	misc.Init()
+	latestExportPostfix := "latest-export"
+	fullExportPostfix := "full-export"
+	tests := []struct {
+		name                 string
+		endpoint             string
+		method               string
+		responseCode         int
+		expectedResponseBody string
```

Comment:
should we add an expected content type here too?

Reply:
added `http.DetectContentType(body)` to compare with.